dlp/api/Snippets/DeidentifyWithDateShift.cs

```diff
@@ -0,0 +1,141 @@
+        var protoRows = csvRows.Select(CsvRow =>
+        {
+            var rowValues = CsvRow.Split(',');
+            var protoValues = rowValues.Select(RowValue =>
+            {
+                if (System.DateTime.TryParse(RowValue, out var parsedDate))
+                {
+                    return new Value
+                    {
+                        DateValue = new Google.Type.Date
+                        {
+                            Year = parsedDate.Year,
+                            Month = parsedDate.Month,
+                            Day = parsedDate.Day
+                        }
+                    };
+                }
+                else
+                {
+                    return new Value
+                    {
+                        StringValue = RowValue
+                    };
+                }
+            });
+
+            var rowObject = new Table.Types.Row();
+            rowObject.Values.Add(protoValues);
+            return rowObject;
+        });
+
+        var dateFieldList = dateFields
+            .Split(',')
+            .Select(field => new FieldId { Name = field });
+
+        // Construct + execute the request
+        var dateShiftConfig = new DateShiftConfig
+        {
+            LowerBoundDays = lowerBoundDays,
+            UpperBoundDays = upperBoundDays
+        };
+        var hasKeyName = !string.IsNullOrEmpty(keyName);
+        var hasWrappedKey = !string.IsNullOrEmpty(wrappedKey);
+        var hasContext = !string.IsNullOrEmpty(contextField);
+        if (hasKeyName && hasWrappedKey && hasContext)
+        {
+            dateShiftConfig.Context = new FieldId { Name = contextField };
+            dateShiftConfig.CryptoKey = new CryptoKey
+            {
+                KmsWrapped = new KmsWrappedCryptoKey
+                {
+                    WrappedKey = ByteString.FromBase64(wrappedKey),
+                    CryptoKeyName = keyName
+                }
+            };
+        }
+        else if (hasKeyName || hasWrappedKey || hasContext)
+        {
+            throw new ArgumentException("Must specify ALL or NONE of: {contextFieldId, keyName, wrappedKey}!");
+        }
+
+        var deidConfig = new DeidentifyConfig
+        {
+            RecordTransformations = new RecordTransformations
+            {
+                FieldTransformations =
+                    {
+                        new FieldTransformation
+                        {
+                            PrimitiveTransformation = new PrimitiveTransformation
+                            {
+                                DateShiftConfig = dateShiftConfig
+                            },
+                            Fields = { dateFieldList }
+                        }
+                    }
+            }
+        };
+
+        var response = dlp.DeidentifyContent(
+            new DeidentifyContentRequest
+            {
+                Parent = $"projects/{projectId}",
```

Comment:
ParentAsProjectName, if you can't do that there's a missing annotation in the proto.